local/local.go

```diff
@@ -71,28 +76,107 @@ func FromBaseImage(imageName string) ImageOption {
 	}
 }
 
-func NewImage(repoName string, dockerClient client.CommonAPIClient, ops ...ImageOption) (*Image, error) {
+func WithPlatform(platform imgutil.Platform) InitialImageOption {
+	return func(i *Image) (*Image, error) {
+		if platform.OS != "" && platform.OS != i.inspect.Os {
+			return nil, fmt.Errorf(`invalid os: platform os "%s" must match the daemon os "%s"`, platform.OS, i.inspect.Os)
+		}
+
+		i.inspect.Architecture = platform.Architecture
+		i.inspect.OsVersion = platform.OSVersion
+
+		i.platform = platform
+
+		return i, nil
+	}
+}
+
+func NewImage(repoName string, dockerClient client.CommonAPIClient, ops ...interface{}) (*Image, error) {
 	var err error
 
-	inspect, err := defaultInspect(dockerClient)
+	defaultPlatform, err := defaultPlatform(dockerClient)
 	if err != nil {
 		return nil, err
 	}
 
+	inspect := defaultInspect(defaultPlatform)
+
 	image := &Image{
 		docker:           dockerClient,
 		repoName:         repoName,
 		inspect:          inspect,
+		platform:         defaultPlatform,
 		layerPaths:       make([]string, len(inspect.RootFS.Layers)),
 		downloadBaseOnce: &sync.Once{},
 	}
 
-	for _, v := range ops {
-		image, err = v(image)
+	image, err = processImageOptions(image, ops)
+	if err != nil {
+		return nil, err
+	}
+
+	image, err = prepareImage(image)
+	if err != nil {
+		return nil, err
+	}
+
+	return image, nil
+}
+
+func processImageOptions(image *Image, ops []interface{}) (*Image, error) {
+	sort.Slice(ops, func(i, _ int) bool {
```

Comment:
Is this to ensure that `WithPlatform` is always applied first?

Reply:
Yes, though as per Emily's [comment](https://github.com/buildpacks/imgutil/pull/91/files/038872784f71cf5d70fce08b92ebf5716337180d#r583956120), I think I'll add another commit to this to move the processing of all the options into here.